components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java

```diff
@@ -930,4 +934,32 @@ private String getTenantDomainFromUserName(AuthenticationContext context, String
         }
         return MultitenantUtils.getTenantDomain(username);
     }
+
+    private boolean isURLContainSensitiveData(HttpServletRequest request, HttpServletResponse response,
+                                              AuthenticationContext context) throws AuthenticationFailedException {
+
+        if (StringUtils.contains(request.getQueryString(), BasicAuthenticatorConstants.USER_NAME) ||
+                StringUtils.contains(request.getQueryString(), BasicAuthenticatorConstants.PASSWORD)) {
+
+            String loginPage = ConfigurationFacade.getInstance().getAuthenticationEndpointURL();
+            String queryParams = context.getContextIdIncludedQueryParams();
+            String redirectURL;
+            String retryParam = "";
+            retryParam = BasicAuthenticatorConstants.AUTH_FAILURE_PARAM + "true" +
+                    BasicAuthenticatorConstants.AUTH_FAILURE_MSG_PARAM + "query.params.contains.user.credentials";
+            redirectURL = loginPage + ("?" + queryParams)
```

Comment:
can't we directly assign this?

Reply:
Fixed with https://github.com/wso2-extensions/identity-local-auth-basicauth/pull/140/commits/61b3d02b181581b0ed93ab1a0b1a800aac7389fb